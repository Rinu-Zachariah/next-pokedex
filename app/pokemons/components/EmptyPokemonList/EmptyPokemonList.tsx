import React from 'react';

interface IProps {
  searchText?: string;
  error?: string
}

const EmptyPokemonList: React.FC<IProps> = ({ searchText, error }) => (
  <div className="flex flex-col items-center justify-center h-96 relative">
    <img 
      src="/assets/snorlax.png"
      alt="No Pokemon found"
      height={250}
      width={250}
      className="mb-6 select-none drop-shadow-xl"
      style={{ filter: 'grayscale(40%)' }}
    />
    <div className="absolute left-[70%] top-[18%] z-10 animate-bounce-slow">
      <span className="text-3xl text-blue-300 ml-1">z</span>
      <span className="text-4xl text-blue-200 ml-1">z</span>
      <span className="text-5xl text-blue-100 ml-1">z</span>
    </div>
    <p className="text-lg font-bold text-gray-500 mb-1">
      {error?.includes("not found") || error?.includes("Not found")
        ? `No Pokemon found matching "${searchText}"`
        : "No Pokemon found!!"}
    </p>
    <p className="text-sm text-gray-400 italic">Try another search or come back later.</p>
  </div>
);

export default EmptyPokemonList;