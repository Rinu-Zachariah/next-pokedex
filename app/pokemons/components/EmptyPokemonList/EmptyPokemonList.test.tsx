import React from 'react';
import { render } from '@testing-library/react';
import EmptyPokemonList from './EmptyPokemonList';

describe('EmptyPokemonList', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<EmptyPokemonList searchText="charizard" error="some error" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
