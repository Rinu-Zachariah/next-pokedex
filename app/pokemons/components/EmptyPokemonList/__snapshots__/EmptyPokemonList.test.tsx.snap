// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyPokemonList renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center h-96 relative"
  >
    <img
      alt="No Pokemon found"
      class="mb-6 select-none drop-shadow-xl"
      height="250"
      src="/assets/snorlax.png"
      style="filter: grayscale(40%);"
      width="250"
    />
    <div
      class="absolute left-[70%] top-[18%] z-10 animate-bounce-slow"
    >
      <span
        class="text-3xl text-blue-300 ml-1"
      >
        z
      </span>
      <span
        class="text-4xl text-blue-200 ml-1"
      >
        z
      </span>
      <span
        class="text-5xl text-blue-100 ml-1"
      >
        z
      </span>
    </div>
    <p
      class="text-lg font-bold text-gray-500 mb-1"
    >
      No Pokemon found!!
    </p>
    <p
      class="text-sm text-gray-400 italic"
    >
      Try another search or come back later.
    </p>
  </div>
</DocumentFragment>
`;
